hulu/roku-dev-cli
# adding reporter for parsing rendezvous logs

This adds a reporter that parses the logrendezvous output from telnet and builds a meaningful report.  The report breaks down:
* most frequent rendezvous (and the average durations)
* longest single rendezvous duration
* adjacent rendezvous (by line) for opportunities to use _queueFields_

Example output:
```
=== Rendezvous by Frequency ===
18 - pkg:/components/some_brs_file.brs(27) (avg. duration: 0.001s)
14 - pkg:/components/some_other_brs_file.brs(178) (avg. duration: 0.023s)
...

=== Rendezvous by Longest Single Duration ===
0.369s - pkg:/components/some_brs_file.brs(24)
0.291s - pkg:/components/some_other_brs_file.brs(178)
...

=== Adjacent Rendezvous (opportunity to use `queueFields`?) ===
pkg:/components/some_brs_file.brs - lines adjacent to other rendezvous: 236-238, 251-258
pkg:/components/some_other_brs_file.brs - lines adjacent to other rendezvous: 45-47
...
```


To test:
* turn on rendezvous output by telnetting onto port 80 and running "logrendezvous on"
* start the roku script with "--report-rendezvous" flag
* observe outputted file
